domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterModuleTest.kt

```diff
@@ -0,0 +1,134 @@
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+
+/**
+ * [PlatformParameterModuleTest] verifies the working of [PlatformParameterModule] by testing
+ * the [PlatformParameterValue] received in different cases
+ * */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterModuleTest.TestApplication::class)
+class PlatformParameterModuleTest {
+
+  @Inject
+  @TestStringParam
+  lateinit var testStringPlatformParameter: PlatformParameter<String>
+
+  @Inject
+  @TestIntegerParam
+  lateinit var testIntegerPlatformParameter: PlatformParameter<Int>
+
+  @Inject
+  @TestBooleanParam
+  lateinit var testBooleanPlatformParameter: PlatformParameter<Boolean>
+
+  private val mockPlatformParameterMap by lazy {
+    val mockStringPlatformParameter = ParameterValue.newBuilder()
+      .setString(TEST_STRING_PARAM_VALUE).build()
+    val mockIntegerPlatformParameter = ParameterValue.newBuilder()
+      .setInteger(TEST_INTEGER_PARAM_VALUE).build()
+    val mockBooleanPlatformParameter = ParameterValue.newBuilder()
+      .setBoolean(TEST_BOOLEAN_PARAM_VALUE).build()
+
+    mapOf(
+      TEST_STRING_PARAM_NAME to mockStringPlatformParameter,
+      TEST_INTEGER_PARAM_NAME to mockIntegerPlatformParameter,
+      TEST_BOOLEAN_PARAM_NAME to mockBooleanPlatformParameter
+    )
+  }
+
+  @Test
+  fun testModule_initPlatformParameterMap_retrieveTestStringParameter_returnsParamValue() {
+    setUpTestApplicationComponent(mockPlatformParameterMap)
+    assertThat(testStringPlatformParameter.value).isEqualTo(TEST_STRING_PARAM_VALUE)
+  }
+
+  @Test
+  fun testModule_doNotInitPlatformParameterMap_retrieveTestStringParameter_returnsDefaultValue() {
+    setUpTestApplicationComponent(mapOf())
+    assertThat(testStringPlatformParameter.value).isEqualTo(TEST_STRING_PARAM_DEFAULT_VALUE)
+  }
+
+  @Test
+  fun testModule_initPlatformParameterMap_retrieveTestIntegerParameter_returnsParamValue() {
+    setUpTestApplicationComponent(mockPlatformParameterMap)
+    assertThat(testIntegerPlatformParameter.value).isEqualTo(TEST_INTEGER_PARAM_VALUE)
+  }
+
+  @Test
+  fun testModule_doNotInitPlatformParameterMap_retrieveTestIntegerParameter_returnsDefaultValue() {
+    setUpTestApplicationComponent(mapOf())
+    assertThat(testIntegerPlatformParameter.value).isEqualTo(TEST_INTEGER_PARAM_DEFAULT_VALUE)
+  }
+
+  @Test
+  fun testModule_initPlatformParameterMap_retrieveTestBooleanParameter_returnsParamValue() {
+    setUpTestApplicationComponent(mockPlatformParameterMap)
+    assertThat(testBooleanPlatformParameter.value).isEqualTo(TEST_BOOLEAN_PARAM_VALUE)
+  }
+
+  @Test
+  fun testModule_doNotInitPlatformParameterMap_retrieveTestBooleanParameter_returnsDefaultValue() {
+    setUpTestApplicationComponent(mapOf())
+    assertThat(testBooleanPlatformParameter.value).isEqualTo(TEST_BOOLEAN_PARAM_DEFAULT_VALUE)
+  }
+
```

Comment:
Also add tests that if you have a partial map it returns defaults or expected value (e.g. if the map only has integer or boolean, it returns those values but requesting a string returns the default string). You probably don't need all ways to do this but a couple to check that you get expected values or expected defaults.

Reply:
Done. Added a single test case after removing the string parameter from the map and testing for its default value while other have the normal value. I guess it is explainatory enough. LMK if we need to add tests for other types too